### ISSUE TITLE
Error when trying to play Vimeo video

### ISSUE THREAD

**Issue by pdehaan** [meandavejustice/min-vid#273]
Built locally using `$ ./node_modules/.bin/jpm run -b Nightly`

I've been playing about 10 videos in the same session from https://www.reddit.com/r/videos and started to see some slowdown after the 7-8th video.

Here's the error when I tried to play https://vimeo.com/channels/musicvideos/180016993 from Vimeo.

```
console.error: min-vid:
JPM [error]   Message: TypeError: resp.json is null
  Stack:
    getVimeoUrl/<.onComplete@resource://gre/modules/commonjs/toolkit/loader.js -> resource://min-vid/lib/get-vimeo-url.js:9:7
emitOnObject@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:112:9
emit@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:89:38
onreadystatechange@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/request.js:104:7
EventHandlerNonNull*runRequest@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/request.js:100:3
Request<.get@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/request.js:145:5
getVimeoUrl@resource://gre/modules/commonjs/toolkit/loader.js -> resource://min-vid/lib/get-vimeo-url.js:6:3
launchVideo@resource://gre/modules/commonjs/toolkit/loader.js -> resource://min-vid/lib/launch-video.js:25:5
init/onMessageVimeoHandler@resource://gre/modules/commonjs/toolkit/loader.js -> resource://min-vid/lib/context-menu-handlers.js:51:5
emitOnObject@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:112:9
emit@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:89:38
workerMessageReceived@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/context-menu.js:455:17
emitOnObject@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:112:9
emit@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:89:38
emitOnObject@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:112:9
emit@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:89:38
emitOnObject@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:123:45
emit@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/event/core.js:89:38
messageReceived@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/remote/parent.js:96:37
```

**Comment by pdehaan** [comment]
I also got this intermittent error from a YouTube video, but was unable to reproduce a second time. not sure why it'd give me `null` for node though:

```
console.error:
JPM [error]   Message: TypeError: node is null
  Stack:
    @javascript:
self.on('click', function (node, data) {
  self.postMessage(node.href || node['data-feed-url']);
});:3:3
onEvent@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/loader/sandbox.js -> resource://gre/modules/commonjs/sdk/content/content-worker.js:45:22
onChromeEvent@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/loader/sandbox.js -> resource://gre/modules/commonjs/sdk/content/content-worker.js:91:16
```

I've tried to reproduce with my previous YouTube video's I've watched and a few others on the /r/videos homepage, but haven't been able to reproduce this error.

I can seemingly get the Vimeo error every time though.

**Comment by meandavejustice** [comment]
I will occasionally get this with certain vimeo videos. I think the solution for now is to have error handling for this case instead of failing silently. We'll try to get this in for `mvp` but since it is intermittent I don't think it is a very high priority for launch.

**Comment by pdehaan** [comment]
> We'll try to get this in for mvp but since it is intermittent I don't think it is a very high priority for launch.

This is failing for me for all (well, all TWO, so crappy sample size) videos currently listed from /r/videos:

- https://vimeo.com/183252171 -- which takes me to a 404 page, so looks like the video was removed. But still a bad experience from /r/videos.
- https://vimeo.com/channels/musicvideos/180016993

It seems to be working for videos on https://vimeo.com homepage.

Although it looks like for each video I try playing (ie: https://vimeo.com/183296320), clicking "Send to mini player" brings up an "Unvalid URI. Load of media resource  failed." message in my Browser Console from resource://min-vid/data/default.html

Everything seems to work as expected from:

- https://www.reddit.com/r/videos/comments/2wifh9/this_short_film_just_became_staff_pick_on_vimeo/ (right click and send to mini player)
- https://vimeo.com/42934682 (right click this link from the Reddit comments page above)

--- 

TL;DR: Yeah, seems like we just need a better error/promise checking and isn't a release blocker.
